lua/image/image.lua

```diff
@@ -114,25 +114,21 @@ function Image:render(geometry)
       end
     end
 
-    -- TODO: chain rerendering only the next affected image after this one
-    -- local next_image = nil
-    -- local next_image_distance = math.huge
-    -- for _, image in pairs(self.global_state.images) do
-    --   -- if image.buffer == self.buffer and image.geometry.y > self.geometry.y then image:render() end
-    --   if image.buffer == self.buffer then
-    --     local distance = image.geometry.y - self.geometry.y
-    --     if distance > 0 and distance < next_image_distance then
-    --       next_image = image
-    --       next_image_distance = distance
-    --     end
-    --   end
-    -- end
-    -- utils.debug(("(image.render) id: %s, next_image: %s"):format(self.id, next_image and next_image.id))
-    -- if next_image then next_image:render() end
-
-    -- rerender any images that are below this one
-    for _, image in pairs(self.global_state.images) do
-      if image.buffer == self.buffer and image.geometry.y > self.geometry.y then image:render() end
+    if self.with_virtual_padding then
+      -- rerender any images that are below this one
+      local buf_imgs = vim.tbl_filter(function(x)
+        return x.buffer == self.buffer
+      end, vim.tbl_values(self.global_state.images))
+      table.sort(buf_imgs, function(a, b)
+        return a.geometry.y < b.geometry.y
+      end)
+      for _, image in ipairs(buf_imgs) do
+        if image.buffer == self.buffer and image.geometry.y > self.geometry.y then
+          image:render()
+
+          if image.with_virtual_padding then break end
```

Comment:
wouldn't this break rendering of the images below that need their position updated?
maybe we can have some kind of rendering queue set to fix the exponential issue.
i'm thinking of the scenario with markdown documents, if there are three images that fit inside the screen and the first one gets its height +1d, the second one would update but the third one wouldn't because of the break (i think?)

Reply:
The second one being rendered and moved would trigger the re-render on the third image I think.

Reply:
Ah yes, I think it would. Checking <3